### ISSUE TITLE
Validate user uploaded SSH public keys

### ISSUE THREAD

**Issue by OmenWild** [ucdavis/hippo#183]
It would be nice to do as much validation of the SSH public key that users upload as possible. There is a good description of the format the the base64 encoded portion here https://www.thedigitalcatonline.com/blog/2018/04/25/rsa-keys/ under the `The OpenSSH public key format` section. The very short version is the first 4 bytes represent the length of the next chunk. The 2nd chunk is `ssh-` and the algorithm. The first part of each of my 3 public keys look like this in hex:
```
0000000 00 00 00 07 73 73 68 2d 72 73 61 00 00 00 03 01

0000000 00 00 00 0b 73 73 68 2d 65 64 32 35 35 31 39 00

0000000 00 00 00 07 73 73 68 2d 72 73 61 00 00 00 03 01
```

And like this in ascii: 
```
0000000 nul nul nul bel   s   s   h   -   r   s   a nul nul nul etx soh

0000000 nul nul nul  vt   s   s   h   -   e   d   2   5   5   1   9 nul

0000000 nul nul nul bel   s   s   h   -   r   s   a nul nul nul etx soh
```

So very basic validation would be that bytes 0-3 are a small integer, and that 4-7  are `73 73 68 2d`.

**Comment by srkirkland** [comment]
@laholstege so basically key looks something like 
```
ssh-rsa AAAAB3NzaC1yc2E......MCEC9wWCJOoo0= postit@QHC9P6H4JJ
```

So to validate like omen suggests, we'd strip off everything upto and including the first space and everything after and including the last space.  That just leaves us with some base64 encoded stuff.  Then we do a simple `Convert.FromBase64String(sshkey);` to grab the bytes and do some checking.  I think the simplest would be to just check that bytes[4] through bytes[7] are `73 73 68 2d`.  which i think you can most easily compare if you use their int conversions of `115 115 104 45` but don't quote me on the exact values.

We'd basically want this done as an extra check method in the ssh service, and then in the react component when creating an account you'd want to check this, either immediately after pasting it in or at least before the new account is created, give a good error message if it fails, etc.

**Comment by tefarley** [comment]
I'd like to add functionality to this. Mainly support for ssh key algos other than rsa. Mostly an edge case here but I wouldn't be able to use hippo to update MY key because I use ed25519 or ecdsa

**Comment by sprucely** [comment]
Addressed by PR #252
